frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts

```diff
@@ -37,41 +36,44 @@ export class SuitableLessonComponent implements OnInit {
         this.selectedDate.setDate(this.selectedDate.getDate() + this.amountOfDayInWeek);
 
         this.setDays();
+
+        this.dateSelected.emit(new Date(this.selectedDate));
     }
 
     weekDec(): void {
         this.selectedDate.setDate(this.selectedDate.getDate() - this.amountOfDayInWeek);
 
         this.setDays();
+
+        if (moment(this.selectedDate).isSameOrAfter(moment(), 'day')) {
+            this.dateSelected.emit(new Date(this.selectedDate));
+        }
     }
 
-    async setDays(): Promise<void> {
+    setDays(): void {
         this.days = [];
-        const promises: Promise<{ date: Date, meetingsCount: number }>[] = [];
 
         for (let i = 0; i < 7; i++) {
             const date = moment(this.selectedDate).add(i - this.selectedDate.getDay() + 1, 'day').toDate();
-            const promise = this.getMeetingCount(date).then(meetingsCount => ({ date, meetingsCount }));
 
-            promises.push(promise);
+            this.days.push({ date, meetingsAmount: 0 });
         }
```

Comment:
What about like
this.days = Array(7).keys().map(el => {
    const date = moment(this.selectedDate).add(el - this.selectedDate.getDay() + 1, 'day').toDate();
    return { date; meetingsAmount: 0 };
})